sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchAsyncClient.java

```diff
@@ -0,0 +1,263 @@
+import com.azure.storage.blob.models.StorageException;
+import com.azure.storage.common.Utility;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiFunction;
+
+import static com.azure.core.implementation.util.FluxUtil.withContext;
+import static com.azure.storage.blob.implementation.PostProcessor.postProcessResponse;
+
+/**
+ * This class provides a client that contains all operations that apply to Azure Storage Blob batching.
+ *
+ * <p>This client offers the ability to delete and set access tier on multiple blobs at once and to submit a {@link
+ * BlobBatch}.</p>
+ *
+ * @see BlobBatch
+ * @see BlobBatchClientBuilder
+ */
+@ServiceClient(builder = BlobBatchClientBuilder.class, isAsync = true)
+public final class BlobBatchAsyncClient {
+    private final AzureBlobStorageImpl client;
+
+    BlobBatchAsyncClient(String accountUrl, HttpPipeline pipeline) {
+        this.client = new AzureBlobStorageBuilder()
+            .url(accountUrl)
+            .pipeline(pipeline)
+            .build();
+    }
+
+    /**
+     * Gets a {@link BlobBatch} used to configure a batching operation to send to Azure Storage blobs.
+     *
+     * @return a new {@link BlobBatch} instance.
+     */
+    public BlobBatch getBlobBatch() {
+        return new BlobBatch(client.getUrl(), client.getHttpPipeline());
+    }
+
+    /**
+     * Submits a batch operation.
+     *
+     * <p>If any request in a batch fails this will throw a {@link StorageException}.</p>
+     *
+     * <p><strong>Code samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.batch.BlobBatchAsyncClient.submitBatch#BlobBatch}
+     *
+     * @param batch Batch to submit.
+     * @return An empty response indicating that the batch operation has completed.
+     * @throws StorageException If any request in the {@link BlobBatch} failed or the batch request is malformed.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> submitBatch(BlobBatch batch) {
+        return withContext(context -> submitBatchWithResponse(batch, true, context)).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Submits a batch operation.
+     *
+     * <p>If {@code throwOnAnyFailure} is {@code true} a {@link StorageException} will be thrown if any request
+     * fails.</p>
+     *
+     * <p><strong>Code samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.batch.BlobBatchAsyncClient.submitBatch#BlobBatch-boolean}
+     *
+     * @param batch Batch to submit.
+     * @param throwOnAnyFailure Flag to indicate if an exception should be thrown if any request in the batch fails.
+     * @return A response only containing header and status code information, used to indicate that the batch operation
+     * has completed.
+     * @throws StorageException If {@code throwOnAnyFailure} is {@code true} and any request in the {@link BlobBatch}
```

Comment:
This should not actually throw an exception. Instead, the error should be propagated through the error channel of the `Mono` . Same for all the APIs here that return a Publisher (Mono/Flux).